URBNSwiftAlert/Classes/AlertViewController.swift

```diff
@@ -0,0 +1,337 @@
+                alertContainer.widthAnchor.constraint(lessThanOrEqualToConstant: minMax.max).isActive = true
+            }
+            else {
+                alertContainer.widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width - alertStyler.alert.horizontalMargin*2).isActive = true
+            }
+        }
+        alertContainer.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        alertViewYContraint = alertContainer.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: 0)
+        alertViewYContraint?.isActive = true
+    }
+    
+    func setVisible(isVisible: Bool, completion: ((Void) -> Void)? = nil) {
+        let scaler: CGFloat = 0.3
+        guard let ac = alertView else {
+            assertionFailure()
+            return
+        }
+        
+        if isVisible {
+            ac.alpha = 0.0
+            ac.transform = CGAffineTransform(scaleX: scaler, y: scaler)
+        }
+        
+        let alpha: CGFloat = isVisible ? 1.0 : 0.0
+        let endingTransform = isVisible ? CGAffineTransform.identity : CGAffineTransform(scaleX: scaler, y: scaler)
+        
+        let bounceAnimation = {
+            ac.transform = endingTransform
+        }
+        
+        let fadeAnimation = { [unowned self] in
+            ac.alpha = alpha
+            if self.alertStyler.blur.isEnabled {
+                self.blurImageView?.alpha = alpha
+            }
+        }
+        
+        if alertStyler.animation.isAnimated {
+            let duration = TimeInterval(alertStyler.animation.duration)
+            let damping = alertStyler.animation.damping
+            let initVel = isVisible ? 0 : alertStyler.animation.initialVelocity
+            
+            UIView.animate(withDuration: duration, delay: 0, usingSpringWithDamping: damping, initialSpringVelocity: initVel, options: [], animations: bounceAnimation, completion: { (complete) in
+                if complete {
+                    completion?()
+                }
+            })
+            
+            UIView.animate(withDuration: duration/2, animations: fadeAnimation)
+        }
+        else {
+            fadeAnimation()
+            bounceAnimation()
+            completion?()
+        }
+    }
+    
+    func addBlurScreenshot(withSize size: CGSize? = nil) {
+        if let screenShot = UIImage.screenShot(view: viewForScreenshot, afterScreenUpdates: true) {
+            let blurredSize = size ?? viewForScreenshot.bounds.size
+            
+            let rect = CGRect(x: 0, y: 0, width: blurredSize.width, height: blurredSize.height)
+            
+            blurImageView = UIImageView(frame: rect)
+            blurImageView?.image = screenShot.withRenderingMode(.alwaysTemplate)
+            blurImageView?.tintColor = alertStyler.blur.tint
+            let blurEffect = UIBlurEffect(style: .light)
+            let blurView = UIVisualEffectView(effect: blurEffect)
```

Comment:
We should be using Apple's blur class, not the built in visual effects. They give really no control over the blur-- we want much less blur. You can control the blur radius & saturationDelta in the current implementation of URBNAlert, id expect the same here.

I am also seeing some weirdness on the dismiss animation, but I think it may be because the blur is so extreme

Reply:
Luckily looks like there is a swift version of the UIImageEffects blur code:

https://github.com/globchastyy/SwiftUIImageEffects/blob/master/Source/UIImageEffects.swift